src/extension.ts

```diff
@@ -96,16 +99,35 @@ function fetchAndLaunchMetals(context: ExtensionContext, javaHome: string) {
   if (dottyArtifact && fs.existsSync(dottyArtifact)) {
     outputChannel.appendLine(
       `Metals will not start since Dotty is enabled for this workspace. ` +
-        `To enable Metals, remove the file ${dottyArtifact} and run 'Reload window'`
+      `To enable Metals, remove the file ${dottyArtifact} and run 'Reload window'`
     );
     return;
   }
 
   outputChannel.appendLine(`Java home: ${javaHome}`);
   const javaPath = path.join(javaHome, "bin", "java");
-  const coursierPath = path.join(context.extensionPath, "./coursier");
 
   const config = workspace.getConfiguration("metals");
+
+  const customCoursier = config.get<string>("coursierLocation");
+
+  var coursierPath = "";
+  if (customCoursier && customCoursier.length > 0) {
+    if (fs.existsSync(customCoursier))
+      coursierPath = customCoursier;
+    else {
+      const msg = "Invalid custom Coursier path - no such file exists, either clear or change it";
+      window.showErrorMessage(msg, openSettingsAction).then(resp => {
+        if (resp === openSettingsAction) {
+          commands.executeCommand(openSettingsCommand);
+        }
+      });
+    }
+  }
+  else {
+    coursierPath = path.join(context.extensionPath, "./coursier");
+  }
```

Comment:
Can you move this logic to a separate utility function and only leave

```
const coursierPath = getCoursierPath()
```

here?